_posts/2021-02-11-tabnet/tabnet.Rmd

```diff
@@ -0,0 +1,411 @@
+
+What she calls interpretability, in contrast, is deeply related to domain knowledge:
+
+> Interpretability is a domain-specific notion [...] Usually, however, an interpretable machine learning model is constrained in model form so that it is either useful to someone, or obeys structural knowledge of the domain, such as monotonicity [e.g.,8], causality, structural (generative) constraints, additivity [9], or physical constraints that come from domain knowledge. Often for structured data, sparsity is a useful measure of interpretability [...]. Sparse models allow a view of how variables interact jointly rather than individually. [...] e.g., in some domains, sparsity is useful,and in others is it not.
+
+If we accept these well-thought-out definitions, what can we say about TabNet? Is looking at attention masks more like constructing a post-hoc model or more like having domain knowledge incorporated? I believe Rudin would argue the former, since
+
+-   the image-classification example she uses to point out weaknesses of explainability techniques employs saliency maps, a technical device comparable, in some ontological sense, to attention masks;
+
+-   the sparsity enforced by TabNet is a technical, not a domain-related constraint;
+
+-   we only know *what* features were used by TabNet, not *how* it used them.
+
+On the other hand, one could disagree with Rudin (and others) about the premises. Do explanations *have* to be modeled after human cognition to be considered valid? Personally, I guess I'm not sure, and to cite from a post by [Keith O'Rourke on just this topic of interpretability](https://statmodeling.stat.columbia.edu/2018/10/30/explainable-ml-versus-interpretable-ml/),
+
+> As with any critically-thinking inquirer, the views behind these deliberations are always subject to rethinking and revision at any time.
+
+In any case though, we can be sure that this topic's importance will only grow with time. While in the very early days of the GDPR, the EU General Data Protection Regulation, it was said that [Article 22](https://www.gdpr.org/regulation/article-22.html) (on automated decision-making) would have significant impact on how ML is used[^4], unfortunately the current view seems to be that its wordings are far too vague to have immediate consequences (e.g., @wachter). But this will be a fascinating topic to follow, from a technical as well as a political point of view.
```

Comment:
```suggestion
In any case though, we can be sure that this topic's importance will only grow with time. While in the very early days of the GDPR (the EU General Data Protection Regulation) it was said that [Article 22](https://www.gdpr.org/regulation/article-22.html) (on automated decision-making) would have significant impact on how ML is used[^4], unfortunately the current view seems to be that its wordings are far too vague to have immediate consequences (e.g., @wachter). But this will be a fascinating topic to follow, from a technical as well as a political point of view.
```